### DISCUSSION TITLE
Can we skip converter type validation if appropriate?

### DISCUSSION THREAD

**Discussion by Ghostbird** [CommunityToolkit/Maui#1833] [General]
MAUI unfortunately doesn't correctly resolve the target type for bindings inside a multi-binding. This means that validated converters cannot be used in those inner bindings, even though they would be dead useful.

In my case it took me a long time to figure out why this didn't work:
```csharp
label.Bind(Label.TextProperty,
  new Binding("Foo", converter: Converter),
  new Binding("Bar")
  convert: ((string?, string?) x) => x.Item1 ?? x.Item2,
  mode: BindingMode.OneWay
);
```
It turned out that this is because the `Converter` is a `BaseConverterOneWay<object?, string>`. This converter is validated, and throws when you attempt to use it in the multi-binding, because the `targetType` will be `object`.
In this case that's annoying, because I know that the `targetType` is not really `object`.

What is the best way to approach this?

- Can it be fixed/avoided in this library?
- Should MAUI be fixed to correctly pass the `targetType`?
- Should I just implement IValueConverter instead?

For now I've wrapped my `Converter` in a loosely typed `FuncConverter` to avoid this problem:
```csharp
label.Bind(Label.TextProperty,
  new Binding("Foo", converter: new FuncConverter<object?, object?, object?>(
    (object? value) => Converter is not null
    ? Converter.Convert(value, typeof(string), null, CultureInfo.CurrentCulture)
    : value
  ),
  new Binding("Bar")
  convert: ((string?, string?) x) => x.Item1 ?? x.Item2,
  mode: BindingMode.OneWay
);
```

**Comment by bijington** [comment]
It might not be exactly what you are looking for but it is possible to disable exceptions from being thrown in the converters with our options:

https://learn.microsoft.com/en-gb/dotnet/communitytoolkit/maui/get-started?tabs=CommunityToolkitMaui#using-the-nuget-packages

**Reply by Ghostbird** [reply]
Thanks for the information. Alas, that's not what I'm looking for. I really want validation exceptions in converters. It's just that this one gets fed the wrong information by MAUI and throws at the wrong time. For now I'll retain the wrapping that widens the type to `object?` and passes validation that way.
